File: parser/src/main/scala/ru/mipt/acsl/decode/parser/DecodeFileType.scala
package ru.mipt.acsl.decode.parser

import javax.swing.Icon

import com.intellij.openapi.fileTypes.LanguageFileType

/**
 * @author Artem Shein
 */
class DecodeFileType extends LanguageFileType(DecodeLanguage.INSTANCE) {
    override def getName: String = "Decode File"

    override def getDescription: String = "Decode device interface description"

    override def getDefaultExtension: String = "decode"

    override def getIcon: Icon = DecodeIcons.FILE
}

object DecodeFileType extends DecodeFileType

File: parser/src/main/java/ru/mipt/acsl/decode/parser/DecodeFileType.java
package ru.mipt.acsl.decode.parser;

import javax.swing.Icon;

import com.intellij.openapi.fileTypes.LanguageFileType;

/**
 * @author Artem Shein
 */
public final class DecodeFileType extends LanguageFileType
{
    public static final DecodeFileType INSTANCE = new DecodeFileType();
    private DecodeFileType() {
        super(DecodeLanguage.INSTANCE);
    }
    @Override public String getName() {
        return "Decode FILE";
    }

    @Override
    public String getDescription() {
        return "Decode device interface description";
    }

    @Override
    public String getDefaultExtension()
    {
        return "decode";
    }

    @Override
    public Icon getIcon()
    {
        return DecodeIcons.FILE;
    }
}
